packages/discord.js/src/structures/ApplicationCommand.js

```diff
@@ -458,6 +453,15 @@ class ApplicationCommand extends Base {
    * {@link ApplicationCommandOptionType.Number} option
    */
 
+  /**
+   * A choice for an application command option.
+   * @typedef {Object} ApplicationCommandOptionChoice
+   * @property {string} name The name of the choice
+   * @property {?string} nameLocalized The localized name of the choice in the provided locale, if any
```

Comment:
Isn't this also possibly undefined?

Reply:
? means undefined or null, but also no

Reply:
No, `?` means null, `[nameLocalized]` means undefined
Are you *sure* it can't be undefined too?

Reply:
[nameLocalized] means optional (aka possibly absent, not present as undefined) and ? means nullable, aka null or undefined
And yes I am sure because the code sets the property to null if not sent by the API

Reply:
Nullable is `T | null`, null or undefined (`T | null | undefined`) is called "nullish".

Reply:
But why does it matter? This will never be undefined and there’s no property on djs that haw undefined as an explicit possible type in the documentation

Reply:
You didn't say that. In this case, `?string` is correct.

Reply:
I did say that, twice